### DISCUSSION TITLE
How to get current page path name with Next js 13

### DISCUSSION THREAD

**Discussion by 1mehdifaraji** [vercel/next.js#43905] [Help] [answered]
I'm trying to find a way to get the current pathname with the new Next js 13 and I tried logging the props for the ```page.tsx``` but it doesn't have a pathname prop : 

```
{ params: {}, searchParams: {} }
```

And since everything is server side , I want to know how can I access the current url pathname from the server side?

**Comment by ciruz** [comment, marked as answer]
Checkout `usePathname` - https://beta.nextjs.org/docs/api-reference/use-pathname , it's for Client Components only, but this is the only thing I'm aware of:

```typescript
'use client';

import { usePathname } from 'next/navigation';

export default function Page() {
  // When URL is /blog/hello, pathname = '/blog/hello'
  // When URL is /dashboard?v=2, pathname = '/dashboard'
  const pathname = usePathname();
  return <div>{pathname}</div>;
}
```

**Reply by nicolaslair** [reply]
That's actually what I am doing - but I find it hard to believe that there is no simple way to get the params id?

**Comment by montanaflynn** [comment]
Seems like a big oversight, I think most people put a special styling on current page in navigation for example.

**Comment by PeterAronZentai** [comment]
What's the equivalent approach for getting the pathname in a Server Component? Detailed info about the request is needed for some generalised logic to be easily implemented. For my particular case I'd like to know the current pathname in layout.tsx - so I can craft the return url part of a login link (if user isn't logged in).

Switching to client component mode for such a trivially available info seems brutal.

**Reply by ciruz** [reply]
There's a trick with a custom middleware to send the requested url as custom header information, but it looks like this also disables SSG permanently. https://github.com/vercel/next.js/issues/43704#issuecomment-1411186664
